src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh

```diff
@@ -85,20 +86,24 @@ __mncaFolder=$__toolsDir/dotnetcli/shared/Microsoft.NETCore.App
 __sharedFxVersion=`ls $__mncaFolder | sed 'r/\([0-9]\+\).*/\1/g' | sort -n | tail -1`
 __sharedFxDir=$__toolsDir/dotnetcli/shared/Microsoft.NETCore.App/$__sharedFxVersion/
 
-case $(uname -s) in
-    Darwin)
-        __packageRid=osx-x64
-        __libraryExtension=dylib
-        ;;
-    Linux)
-        __packageRid=linux-x64
-        __libraryExtension=so
-        ;;
-    *)
-        echo "Unsupported OS $(uname -s) detected. Skipping crossgen of the toolset."
-        exit 0
-        ;;
-esac
+# TODO: once all call-sites start passing <rid> argument to this script
+#       remove case block and replace with error and exit statements
+if [ -z "$__packageRid" ]; then
+    case $(uname -s) in
+        Darwin)
+            __packageRid=osx-x64
+            __libraryExtension=dylib
+            ;;
+        Linux)
+            __packageRid=linux-x64
+            __libraryExtension=so
+            ;;
+        *)
+            echo "Unsupported OS $(uname -s) detected. Skipping crossgen of the toolset."
+            exit 0
+            ;;
+    esac
+fi
```

Comment:
You need to also initialize the `__libraryExtension` property. Just put it after this step and set it to be `dylib` for if `__packageRid` is `osx-x64` otherwise set it to `so`.

Reply:
Done, set `_libraryExtension` to `so`  by default that will be overwritten in case of OSX.

Reply:
If someone passes in osx-x64 as the rid we still need to set the extension to `dylib` so I think it should be done outside of the check here.

Reply:
Didn't thought of that case. Fixed. :)